sklearn/preprocessing/discretization.py

```diff
@@ -0,0 +1,228 @@
+#title          :discretization.py
+#description    :Implementation of a few discretization algorithms
+#author         :Henry Lin <hlin117@gmail.com>
+#==============================================================================
+
+from __future__ import division
+from math import log
+import numpy as np
+from scipy.stats import entropy
+from sklearn.base import BaseEstimator
+
+class MDLP(BaseEstimator):
+    """Implements the MDLP discretization criterion from Usama Fayyad's
+    paper "Multi-Interval Discretization of Continuous-Valued Attributes
+    for Classification Learning"
+    """
+
+    def __init__(self, **params):
+        self.set_params(**params)
+
+    def get_params(self, deep=True):
+        return {"continuous_columns": self.continuous_columns,
+                "min_depth": self.min_depth}
+
+    def set_params(self, **params):
+        self.continuous_columns = params.get("continous_columns")
+        self.min_depth = params.get("min_depth", 0)
+        return self
+
+    def fit(self, X, y):
+        """Finds the intervals of interest from the input data.
+        """
+        if type(X) is list:
+            X = np.array(X)
+        if type(y) is list:
+            y = np.array(y)
+        if self.continuous_columns is None:
+            self.continuous_columns = range(X.shape[1])
```

Comment:
The initial parameters should not be modified. If this is necessary, store `continuous_columns_`.